File: lizhiqiang/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==0) return nums[0];
        
        int res=nums[0];
        int curSum = nums[0];
        for(int i=1;i<nums.size();i++)
        {
            curSum = max(curSum+nums[i],nums[i]);
            res = max(curSum, res);
        }
        return res;
    }
};

File: wuzhentao/maximumsubarray.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var sum = 0
        var j = Int.min
        
        nums.forEach { (value) in
            sum += value
            j = max(j, sum)
            
            if sum < 0 {
                sum = 0
            }
        }
        
        return j
        
    }
}


let s = Solution()
s.maxSubArray([-1])
